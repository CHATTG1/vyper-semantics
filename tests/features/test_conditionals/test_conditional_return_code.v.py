@public
def foo(i: bool) -> num:
    if i:
        return 5
    else:
        return 7
    return 11
