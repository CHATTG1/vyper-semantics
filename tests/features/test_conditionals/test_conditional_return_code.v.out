$%seq ( $return ( 0 , $lll ( $%seq ( $%seq ( $mstore ( 28 , $calldataload ( 0 ) ) , $mstore ( 32 , 1461501637330902918203684832716283019655932542976 ) , $mstore ( 64 , 170141183460469231731687303715884105727 ) , $mstore ( 96 , -170141183460469231731687303715884105728 ) , $mstore ( 128 , 1701411834604692317316873037158841057270000000000 ) , $mstore ( 160 , -1701411834604692317316873037158841057280000000000 ) ) , $if ( $eq ( $mload ( 0 ) , 1163228536 ) , $%seq ( $calldatacopy ( 320 , 4 , 32 ) , $assert ( $iszero ( $callvalue ) ) , $uclamplt ( $calldataload ( $add ( 4 , 0 ) ) , 2 ) , $if ( $mload ( 320 ) , $%seq ( $%seq ( $mstore ( 0 , 5 ) , $return ( 0 , 32 ) ) ) , $%seq ( $%seq ( $mstore ( 0 , 7 ) , $return ( 0 , 32 ) ) ) ) , $%seq ( $mstore ( 0 , 11 ) , $return ( 0 , 32 ) ) , $stop ) ) ) , 0 ) ) )
